maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/CheckDependencySnapshotsPhase.java

```diff
@@ -309,7 +309,8 @@ private static boolean checkArtifact( Artifact artifact, ReleaseDescriptor relea
         // We are only looking at dependencies external to the project - ignore anything found in the reactor as
         // it's version will be updated
         boolean result = artifact.isSnapshot()
-                && !artifact.getBaseVersion().equals( releaseDescriptor.getProjectOriginalVersion( versionlessKey ) );
+                && !artifact.getBaseVersion().equals( releaseDescriptor.getProjectOriginalVersion( versionlessKey ) )
+                        && releaseDescriptor.getDependencyReleaseVersion( versionlessKey ) == null;
```

Comment:
I see an issue here: users can pass a SNAPSHOT here and bypass the check. This should not be allowed, so let's find the proper position to do this.

Reply:
Hi Robert, how about adding `releaseDescriptor.getDependencyReleaseVersion(versionlessKey).indexOf( Artifact.SNAPSHOT_VERSION ) != -1`. Would that be OK?

Reply:
Hi Robert, I added the extra check.